### PR TITLE
Continue refactoring

### PR DESCRIPTION
**Pull request by zaibon**
#### What this PR resolves:
- Improve logging of AYS
- give duration of a Job
- add some error handling

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,11 +3,6 @@ sudo: enabled
 dist: trusty
 group: edge
 
-# whitelist master branch only
-branches:
-  only:
-    - master
```
Comment (ahussein) [inline .travis.yml]: why did you remove this ?
This means that we will have a new build trigger on every commit on every branch!!

**Comment by ahussein** [comment]
No we dont do that. This is an overkill.
Using the whitelist allow us to run the tests in the following scenarios:
- A commit on master (basically merges from branches)
- On Pull requests

Every time you commit a new change to your PR then a CI build will be triggered and tests will run.

We cannot run a new build each commit, that will make everything slower. Changes will be be tested before they are merged to master once you create a PR.

**Comment by ahussein** [comment]
I will revert that change. If you want to test a specific change without creating a PR then you should test it locally. You can check who to run the tests using the testrunner docs from here: https://github.com/Jumpscale/ays9/tree/master/ays_testrunner

**Comment by zaibon** [comment]
> I will revert that change. If you want to test a specific change without creating a PR then you should test it locally. You can check who to run the tests using the testrunner docs from here: https://github.com/Jumpscale/ays9/tree/master/ays_testrunner

This is not working, we tried with Reem to execute the test locally, impossible.

**Comment by ahussein** [comment]
Then we need to make sure that everyone can run the tests locally. I will take care of that.
